guardrails/guard.py

```diff
@@ -1014,6 +1014,8 @@ def use(
             on: The part of the LLM request to validate. Defaults to "output".
         """
         hydrated_validator = get_validator(validator, *args, **kwargs)
+        if on == "messages":
```

Comment:
this is a way better approach for now than half implementing the 6x changes. it does mean the error strings will ref msg_history but i dont think thats the end of the world for the short term since this does set the interface